README.md

```diff
@@ -2358,33 +2351,39 @@ Docker image:
 
 ### Cross-compilation
 
-Users have been reporting some degree of success using cross compiles
-from Linux x86 host to embedded ARM Linux devices.
+Cross-compilation is possible when using the CMake build system. First you have
+to build and install flatcc for the build architecture. Then when cross-compiling
+provide the install location of the build architecture flatcc package via
+environment variable `flatccCli_ROOT` or via cmake variable `flatccCli_ROOT`.
+
+The tests cannot be run because they are compiled for the host architecture and
+thus cannot run on the build machine, so build with cmake option `FLATCC_TEST` disabled.
+The option `FLATCC_RTONLY` will disable tests and only build the runtime library.
+It is highly recommended to at least run the `tests/monster_test`
+project on a new platform.
```

Comment:
> It is highly recommended to at least run the `tests/monster_test`
project on a new platform.

True, but exactly what should the user do to achieve that?

Reply:
I left that in from the original cross-building section.
User could cross-compile the samples directory. However to do so some he/she has to make some changes to the CMakeLists.txt in the samples directory.

Reply:
We could also have a cross build example project?

Reply:
For example, to (cross-)compile only the monster sample, you have to insert following lines on top of the corresponding CMakeLists.txt:

```cmake
cmake_minimum_required (VERSION 2.8.12)
project(flatcc_samples_monster)

find_package(flatcc)
include(FlatccGenerateSources)
```

If you want to add that code permanently, it should be something like:

```cmake
cmake_minimum_required (VERSION 2.8.12)
project(flatcc_samples_monster)

get_directory_property(hasParent PARENT_DIRECTORY)
if(hasParent)
    find_package(flatcc)
    include(FlatccGenerateSources)
endif()
```

Reply:
I was thinking about adding a directory parallel to monster that builds independently of the main CMake project and which can cross compile and somehow document that sensibly? Not sure what to call it though.

Reply:
And perhaps have a driver script so I go enter ubuntu x64 and cross build to arm or fail due to missing tooling? Could even be a docker container. Just thinking out loud.

Reply:
Well, that is basically what Conan does....